march_hardware/include/march_hardware/joint.h

```diff
@@ -64,16 +64,17 @@ class Joint
   double getVelocityIUAbsolute();
   double getVelocityIUIncremental();
   float getTemperature();
-  IMotionCubeState getIMotionCubeState();
+  MotorControllerState getMotorControllerState();
+  virtual bool checkMotorControllerState(std::ostringstream& error_stream);
 
   std::string getName() const;
   int getTemperatureGESSlaveIndex() const;
-  int getIMotionCubeSlaveIndex() const;
+  int getMotorControllerSlaveIndex() const;
   int getNetNumber() const;
 
   ActuationMode getActuationMode() const;
 
-  bool hasIMotionCube() const;
+  bool hasMotorController() const;
```

Comment:
May be you could return the typ and `nullptr` if there is none.

Reply:
I considered something like that, but it didn't seem very useful anywhere in the code, so I thought hasMotorController would suffice just as well (and it is easier to implement ;))